community/tutorials/gdnative/gdnative-c-example.rst

```diff
@@ -0,0 +1,328 @@
+
+::
+
+    [general]
+
+    singleton=false
+    load_once=true
+    symbol_prefix="godot_"
+
+    [entry]
+
+    X11.64="res://bin/libsimple.so"
+    Windows.64="res://bin/libsimple.dll"
+    OSX.64="res://bin/libsimple.dylib"
+
+    [dependencies]
+
+    X11.64=[]
+    Windows.64=[]
+    OSX.64=[]
+
+This file contains 3 sections.
+
+The **general** section contains some info that tells Godot how to use our module.
+
+If singleton is true our library is automatically loaded and a function called godot_singleton_init is called. We'll leave that for another tutorial.
+
+If load_once is true our library is loaded only once and each individual script that uses our library will use the same data. Any variable you define globally will be accessible from any instance of your object you create. If load_once is false a new copy of the library is loaded into memory each time a script access the library.
+
+The symbol_prefix is a prefix for our core functions. So the godot in godot_nativescript_init for instance. If you use multiple GDnative libraries that you wish to statically link you'll have to use different prefixes. This again is a subject to dive into deeper in a separate tutorial, it is only needed at this time for deployment to iOS as this platform does not like dynamic libraries.
+
+The **entry** section tells us for each platform and feature combination which dynamic library has to be loaded. This also informs the exporter which files need to be exported when exporting to a specific platform.
+
+The **dependencies** section tells Godot what other files need to be exported for each platform in order for our library to work. Say that your GDNative module uses another DLL to implement functionality from a 3rd party library, this is where you list that DLL.
+
+Putting it all together
+-----------------------
+Now that we should have a working GDNative library it is time to fire up Godot and use it. Open up the sample project if you haven't left it open after creating the project all the way at the beginning of this tutorial.
+
+Creating our GDNS file
+----------------------
+With our GDNLIB file we've told Godot how to load our library, now we need to tell it about our "Simple" object class. This we do by creating a GDNS resource file.
+
+Start by clicking the create resource button in the Inspector:
+
+.. image:: img/new_resource.gif
+
+And select NativeScript:
+
+.. image:: img/nativescript_resource.png
+
+Press Create, now the inspector will show a few fields we need to enter. In Class Name we enter "SIMPLE" which is the object class name we used in our C source when calling godot_nativescript_register_class. We also need to select our GDNLIB file by clicking on Library and selecting Load:
```

Comment:
Crash without also setting "SIMPLE" as the resource name,
took some time to figure that out :-(

![grafik](https://user-images.githubusercontent.com/4047289/33797509-69b7dd68-dd09-11e7-95ac-48a0b6bc2690.png)

Reply:
That shouldn't be that way.. 🤔